src/box/lua/ctl.c

```diff
@@ -185,6 +187,26 @@ lbox_ctl_wal_sync(struct lua_State *L)
 	return 0;
 }
 
+static int
+lbox_ctl_deactivate_replica(struct lua_State *L)
+{
+	const char *uuid_str = luaT_checkstring(L, 1);
+	struct tt_uuid uuid;
+	if (tt_uuid_from_string(uuid_str, &uuid) != 0)
+		luaT_error(L);
+	struct replica *replica = replica_by_uuid(&uuid);
+	if (replica == NULL)
+		luaL_error(L, "Cannot deactivate replica: does not exist");
+	if (tt_uuid_is_equal(&uuid, &INSTANCE_UUID))
+		luaL_error(L, "Cannot deactivate replica: deactivation of "
+			   "self is not allowed");
+	if (relay_get_state(replica->relay) == RELAY_FOLLOW)
+		luaL_error(L, "Cannot deactivate replica: it is connected");
```

Comment:
Please use `diag_set` with a proper error code. Better move these checks to `replica_deactivate`.

Reply:
> Please use diag_set with a proper error code.

Didn't find any matching error codes. Anyway, it's a Lua function for database administrator that will be actually used manually - does it worth adding a new error code?

Reply:
> Let's not introduce new error code, we already have plenty of them

Is this a problem?

> nobody parses them in code

Where this conclusion comes from?

> We can use smth like `box.error.ILLEGAL_PARAMS`

We are going to move this checks to the internal `replica_deactivate` function. In this case, `ILLEGAL_PARAMS` doesn't seem appropriate.

Reply:
> Where this conclusion comes from?

We discussed that with @Gerold103 and Vladislav Grubov in the scope of timeout option for synchronous transactions. It was said, that we have 300 error codes and nobody parses them. Just retry in case of any error and that's it

> We are going to move this checks to the internal replica_deactivate function.

Why? We already have assertion there

Reply:
> Let's not introduce new error code, we already have plenty of them and nobody parses them in code. We can use smth like `box.error.ILLEGAL_PARAMS` here, as connected replica is actually an illegal parameter

It isn't an illegal parameter. All function parameters are legal. The replica can't be deactivated because it's busy.



> We discussed that with @Gerold103 and Vladislav Grubov in the scope of timeout option for synchronous transactions.

Having different kinds of timeout errors seems to be an overkill, I agree. In this case I don't see an error we could reuse.

Reply:
Added a special error code `ER_REPLICA_DEACTIVATE` for the function. The error will be updated when we will choose a new name for `box.ctl.deactivate_replica`.